custom_components/frigate/views.py

```diff
@@ -208,9 +210,36 @@ def _create_path(self, path: str, **kwargs: Any) -> str | None:
             return f"api/events/{event_id}/clip.mp4"
         return None
 
-    def _permit_request(self, request: web.Request, config_entry: ConfigEntry) -> bool:
+    def _permit_request(self, request: web.Request, config_entry: ConfigEntry, **kwargs: Any) -> bool:
         """Determine whether to permit a request."""
-        return bool(config_entry.options.get(CONF_NOTIFICATION_PROXY_ENABLE, True))
+
+        is_notification_proxy_enabled = bool(config_entry.options.get(CONF_NOTIFICATION_PROXY_ENABLE, True))
+
+        # If proxy is disabled, immediately reject
+        if not is_notification_proxy_enabled:
+            return False
+
+        notification_expiration_mins = int(config_entry.options.get(CONF_EXPIRE_NOTIFICATIONS_AFTER_MINS, 0))
```

Comment:
Is this `int` cast necessary?

Reply:
We seem to cast the output of `config_entry.options.get` every time it is used. I believe without this cast, the interpreter sets is as an Any | None type.